zirgen/circuit/predicates/predicates.h

```diff
@@ -88,19 +88,35 @@ struct Assumption {
   Assumption() = default;
   // Construct via reading from a stream
   Assumption(llvm::ArrayRef<Val>& stream, bool longDigest = false);
-  // Write to an output
-  void write(std::vector<Val>& stream);
   // Digest into a single value
   DigestVal digest();
 
   DigestVal claim;
   DigestVal controlRoot;
 };
 
+struct UnionClaim {
+  constexpr static size_t size = 2 * kDigestHalfs;
+  // Default constructor
+  UnionClaim() = default;
+  // Construct via reading from a stream
+  UnionClaim(llvm::ArrayRef<Val>& stream, bool longDigest = false);
```

Comment:
I believe 'longDigest' is only every used during lift, which doesn't make sense for a Union Claim?  Definitely the current code never calls this constructor at all...

Reply:
I've removed this constructor 👍